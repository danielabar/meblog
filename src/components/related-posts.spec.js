import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"

import RelatedPosts from "./related-posts"

describe("RelatedPosts", () => {
  it("Renders a card for each post", () => {
    const related = {
      edges: [
        {
          node: {
            id: "nodeid1",
            fields: {
              slug: "/blog/related-post-1/",
            },
            frontmatter: {
              title: "Related Post 1",
              featuredImage: {
                childImageSharp: {
                  gatsbyImageData: {
                    BackgroundColor: "#d8c8d8",
                    height: 600,
                    images: {
                      fallback: {
                        src:
                          "/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg",
                        srcSet:
                          "/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-1-image.jpg 200w,\n/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-1-image.jpg 400w,\n/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg 800w",
                        sizes: "(min-width: 800px) 800px, 100vw",
                      },
                      sources: [
                        {
                          srcSet:
                            "/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-1-image.webp 200w,\n/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-1-image.webp 400w,\n/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-1-image.webp 800w",
                          type: "image/webp",
                          sizes: "(min-width: 800px) 800px, 100vw",
                        },
                      ],
                    },
                    layout: "constrained",
                    width: 800,
                  },
                },
              },
            },
          },
        },
        {
          node: {
            id: "nodeid2",
            fields: {
              slug: "/blog/related-post-2/",
            },
            frontmatter: {
              title: "Related Post 2",
              featuredImage: {
                childImageSharp: {
                  gatsbyImageData: {
                    BackgroundColor: "#d8c8d8",
                    height: 600,
                    images: {
                      fallback: {
                        src:
                          "/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg",
                        srcSet:
                          "/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-2-image.jpg 200w,\n/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-2-image.jpg 400w,\n/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg 800w",
                        sizes: "(min-width: 800px) 800px, 100vw",
                      },
                      sources: [
                        {
                          srcSet:
                            "/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-2-image.webp 200w,\n/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-2-image.webp 400w,\n/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-2-image.webp 800w",
                          type: "image/webp",
                          sizes: "(min-width: 800px) 800px, 100vw",
                        },
                      ],
                    },
                    layout: "constrained",
                    width: 800,
                  },
                },
              },
            },
          },
        },
      ],
    }
    const container = render(<RelatedPosts related={related} />)
    expect(container).toMatchSnapshot()
  })
})
