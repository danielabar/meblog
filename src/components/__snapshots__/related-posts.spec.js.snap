// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedPosts Renders a card for each post 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="related-posts"
      >
        <h2>
          You may also like...
        </h2>
        <div>
          <a
            href="/blog/related-post-1/"
          >
            <div>
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 800px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='600' width='800' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <picture>
                  <source
                    data-srcset="/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-1-image.webp 200w,/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-1-image.webp 400w,/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-1-image.webp 800w"
                    sizes="(min-width: 800px) 800px, 100vw"
                    type="image/webp"
                  />
                  <img
                    alt="Related Post 1"
                    data-gatsby-image-ssr=""
                    data-main-image=""
                    data-src="/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg"
                    data-srcset="/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-1-image.jpg 200w,/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-1-image.jpg 400w,/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg 800w"
                    decoding="async"
                    loading="lazy"
                    sizes="(min-width: 800px) 800px, 100vw"
                    style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                  />
                </picture>
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
                </script>
              </div>
              <p>
                Related Post 1
              </p>
            </div>
          </a>
          <a
            href="/blog/related-post-2/"
          >
            <div>
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 800px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='600' width='800' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <picture>
                  <source
                    data-srcset="/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-2-image.webp 200w,/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-2-image.webp 400w,/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-2-image.webp 800w"
                    sizes="(min-width: 800px) 800px, 100vw"
                    type="image/webp"
                  />
                  <img
                    alt="Related Post 2"
                    data-gatsby-image-ssr=""
                    data-main-image=""
                    data-src="/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg"
                    data-srcset="/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-2-image.jpg 200w,/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-2-image.jpg 400w,/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg 800w"
                    decoding="async"
                    loading="lazy"
                    sizes="(min-width: 800px) 800px, 100vw"
                    style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                  />
                </picture>
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
                </script>
              </div>
              <p>
                Related Post 2
              </p>
            </div>
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="related-posts"
    >
      <h2>
        You may also like...
      </h2>
      <div>
        <a
          href="/blog/related-post-1/"
        >
          <div>
            <div
              class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
              data-gatsby-image-wrapper=""
              style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
            >
              <div
                style="max-width: 800px; display: block;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,%3Csvg height='600' width='800' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                  style="max-width: 100%; display: block; position: static;"
                />
              </div>
              <div
                aria-hidden="true"
                data-placeholder-image=""
                style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
              <picture>
                <source
                  data-srcset="/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-1-image.webp 200w,/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-1-image.webp 400w,/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-1-image.webp 800w"
                  sizes="(min-width: 800px) 800px, 100vw"
                  type="image/webp"
                />
                <img
                  alt="Related Post 1"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg"
                  data-srcset="/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-1-image.jpg 200w,/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-1-image.jpg 400w,/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-1-image.jpg 800w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 800px) 800px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
              </picture>
              <noscript />
              <script
                type="module"
              >
                const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
              </script>
            </div>
            <p>
              Related Post 1
            </p>
          </div>
        </a>
        <a
          href="/blog/related-post-2/"
        >
          <div>
            <div
              class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
              data-gatsby-image-wrapper=""
              style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
            >
              <div
                style="max-width: 800px; display: block;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,%3Csvg height='600' width='800' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                  style="max-width: 100%; display: block; position: static;"
                />
              </div>
              <div
                aria-hidden="true"
                data-placeholder-image=""
                style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
              <picture>
                <source
                  data-srcset="/static/41cac534f5048f26531ee516516637b1/b5535/prelated-post-2-image.webp 200w,/static/41cac534f5048f26531ee516516637b1/f5c71/prelated-post-2-image.webp 400w,/static/41cac534f5048f26531ee516516637b1/0d27e/prelated-post-2-image.webp 800w"
                  sizes="(min-width: 800px) 800px, 100vw"
                  type="image/webp"
                />
                <img
                  alt="Related Post 2"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg"
                  data-srcset="/static/41cac534f5048f26531ee516516637b1/90ed1/prelated-post-2-image.jpg 200w,/static/41cac534f5048f26531ee516516637b1/2070e/prelated-post-2-image.jpg 400w,/static/41cac534f5048f26531ee516516637b1/d0b9c/prelated-post-2-image.jpg 800w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 800px) 800px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
              </picture>
              <noscript />
              <script
                type="module"
              >
                const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
              </script>
            </div>
            <p>
              Related Post 2
            </p>
          </div>
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
